sdk/resourcemanager/Azure.ResourceManager/src/Resources/Custom/ProviderCollection.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.ResourceManager.Core;
+
+namespace Azure.ResourceManager.Resources
+{
+    /// <summary>
+    /// A class representing collection of resources and their operations over their parent.
+    /// </summary>
+    public partial class ProviderCollection : ArmCollection
+    {
+        internal string TryGetApiVersion(ResourceType resourceType, CancellationToken cancellationToken = default)
+        {
+            string version;
+            Dictionary<string, string> resourceVersions;
+            if (!ClientOptions.ResourceApiVersions.TryGetValue(resourceType.Namespace, out resourceVersions))
```

Comment:
Should we move the ResourceApiVersions out of ClientOptions and into ProviderCollection ? I feel a bit uncomfortable having a mutable concurrent dictionary on client options.

Reply:
There is an optimization we will do after GA to make this cache shared across instances that are constructed with the default options that Ted mentioned